generator/src/main/java/com/scottlogic/deg/generator/guice/DataSetWriterProvider.java

```diff
@@ -3,26 +3,38 @@
 import com.google.inject.Inject;
 import com.google.inject.Provider;
 import com.scottlogic.deg.generator.generation.GenerationConfigSource;
-import com.scottlogic.deg.generator.outputs.datasetwriters.CsvDataSetWriter;
-import com.scottlogic.deg.generator.outputs.datasetwriters.DataSetWriter;
-import com.scottlogic.deg.generator.outputs.datasetwriters.MultiDataSetWriter;
-import com.scottlogic.deg.generator.outputs.datasetwriters.SourceTracingDataSetWriter;
+import com.scottlogic.deg.generator.outputs.datasetwriters.*;
 
 public class DataSetWriterProvider implements Provider<DataSetWriter> {
     private final GenerationConfigSource configSource;
+    private final CsvDataSetWriter csvWriter;
+    private final JsonDataSetWriter jsonWriter;
 
     @Inject
-    public DataSetWriterProvider(GenerationConfigSource configSource) {
+    public DataSetWriterProvider(GenerationConfigSource configSource, CsvDataSetWriter csvWriter, JsonDataSetWriter jsonWriter) {
```

Comment:
This feels wrong. 
Will we ever have both CSV and JSON DataSetWriters?
we should probably have a parent class for them.

Reply:
Both writers implement the same interface `DataSetWriter`, this pattern allows them to be created by IoC and then selected as appropriate.

Reply:
Updated #410 to cover review of Provider architecture to allow providers like these to scale in future.